### ISSUE TITLE
amp-accordion: Missing heading markup once rendered

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31155]
## WCAG Level
Level A

## Priority
Medium

## Type
Structure

## Pages/screens/components affected
* [amp-accordion](https://amp.dev/documentation/components/amp-accordion/)

## Description
When content is presented in a way that implies it has a particular structure, like headings that break up sections of content, but the same structure is not there in the HTML, it makes it difficult for anyone who cannot see the intended structure to navigate and understand it.

The structure for an `amp-accordion` expects an HTML heading for each section. When rendered, this heading is turned into a button to trigger the expand/collapse behaviour. As a result, the heading is not structurally conveyed as a heading anymore.

## User impact
Screen readers depend on HTML for information about the structure of content. Working in conjunction with the browser, the screen reader takes information from the HTML and uses it to inform people who cannot see the content what they're dealing with. It will let the user know there is a heading and what level it is. If the HTML doesn't provide this information, a screen reader user will not know that the content is presented in a particular format and this makes it much harder for them to navigate and understand it.

## Required solution
Currently, the heading element inside each accordion `<section>` is turned into a button through the addition of `role="button"`, which overrides its heading semantics.

![One of the example accordions, showing 'Section 1', 'Section 2', 'Section 3' headings, with the first section expanded to show further content](https://user-images.githubusercontent.com/54933393/99181020-8003bf80-2723-11eb-8c70-5bf6e0ad988f.png)

```
<h2 class="i-amphtml-accordion-header" role="button" aria-controls="my-accordion_AMP_content_0" aria-expanded="true" tabindex="0">Section 1</h2>
```

Instead of overriding the heading's role and turning it into a button, we recommend instead to keep the outer heading and to generate a `<button>` element inside it.

```
<h2 class="i-amphtml-accordion-header">
  <button type="button" aria-controls="my-accordion_AMP_content_0" aria-expanded="true">
    Section 1
  </button>
</h2>
```

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Check that heading markup is used to identify text that acts as a heading for a section of content.
2. Check that, once the accordion is rendered, the headings are still present (and exposed to screen readers as heading), with a `<button>` element inside that acts as the actual expand/collapse trigger.

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website  has been tested for the same issue.
* All issues identified throughout the rest of the website  have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 1.3.1 Info and Relationships (Level A)](https://www.w3.org/TR/WCAG21/#info-and-relationships)

## More information
* [WebAIM: Semantic structure](https://webaim.org/techniques/semanticstructure/)
* [Tink: Understanding semantics](https://tink.uk/understanding-semantics/)
* [Nomensa: How to use HTML headings](https://www.nomensa.com/blog/2010/how-use-html-headings)

## Test data
Test date: November 2020
